src/index.ts

```diff
@@ -181,59 +200,50 @@ client.on('voiceStateUpdate', async (oldState, newState) => {
   }
 });
 
-enum PresenceState {
-  IN_CHAMP_SELECT = 'In Champion Select',
-  IN_GAME = 'In Game',
-  IN_LOBBY = 'In Lobby',
-  IN_QUEUE = 'In Queue'
-}
-
 // Event triggered when a user's presence (e.g. status, activity, etc.) is changed.
 client.on('presenceUpdate', async (_, newPresence) => {
-  // Make sure this update is for someone present in the voice channel
+  // Make sure this update is for someone present in the same VoiceChannel as the bot
   const member = newPresence.member;
-  if (member && channel && member.voice.channelId == channel.id) {
-    // Determine if there are any new activities to report
-    const newActivity = newPresence.activities[0];
-    if (newActivity) {
-      // Determine if this presence update is because a user is now playing a game
-      if (newActivity.type === ActivityType.Playing) {
-        // Make sure the game is League of Legends
-        if (newActivity.name === 'League of Legends') {
-          // Determine the state of the activity
-          console.log(newActivity.state);
-          switch (newActivity.state) {
-            case PresenceState.IN_LOBBY:
-            case PresenceState.IN_QUEUE:
-            case PresenceState.IN_CHAMP_SELECT: {
-              break;
-            }
-            case PresenceState.IN_GAME: {
-              if (leagueOfLegendsPollIntervalId === null && IS_LOL_ANNOUNCER_ENABLED) {
-                console.log('Polling game...');
-                leagueOfLegendsPollIntervalId = pollCurrentGame(
-                  channel,
-                  audioPlayer,
-                  PATH_TO_CLIPS
-                );
-              }
-              break;
-            }
-            default: {
-              if (leagueOfLegendsPollIntervalId !== null) {
-                clearInterval(leagueOfLegendsPollIntervalId);
-                console.log('No longer in game, polling stopped...');
-                leagueOfLegendsPollIntervalId = null;
-              }
-              break;
-            }
-          }
+  if (
+    member &&
+    channel &&
+    member.voice.channelId == channel.id &&
+    presenceIndicatesPlayingLeagueOfLegends(newPresence)
+  ) {
+    // Determine the state of the activity
+    const activity = newPresence.activities[0];
```

Comment:
Could there be multiple new activities that end up unhandled here or this would just fire again for each one?

Reply:
Unfortunately I haven't been able to find anything really about when a presence would report multiple activities, so not 100% sure. This works for now but it definitely could be prone to bugs down the line

Reply:
No worries I'm all good with this as is for now